pkg/standalone-utils/contracts/relayer/VaultActions.sol

```diff
@@ -317,7 +318,9 @@ abstract contract VaultActions is IBaseRelayerLibrary {
         private
         returns (bytes memory)
     {
-        if (kind == PoolKind.WEIGHTED) {
+        if (BasePoolUserData.isRecoveryModeExitKind(userData)) {
```

Comment:
Can we add a small comment saying that this check should always go first? Otherwise the weighted / stable encoders will revert.